### PR TITLE
Add serialization tests and code cleanup

### PR DESCRIPTION
**Pull request by oschwald**
This adds tests for serialization of all the response classes. I also reorganized the constructor params added in #52 to be less random and updated some `toString()` methods that were missing fields.

### CODE REVIEW

File: src/test/java/com/maxmind/geoip2/model/JsonTest.java [hunk 1]
```diff
@@ -1,112 +1,304 @@
...
-            + "          \"en\" : \"Minneapolis\"\n"
-            + "       }\n"
-            + "    }\n"
-            + " }\n";
+    @Test
+    public void testInsightsSerialization() throws IOException {
+        String json = JSON.std
+                .composeString()
+                .startObject()
+                .startObjectField("maxmind")
```
Comment (eilara) [inline src/test/java/com/maxmind/geoip2/model/JsonTest.java]: This is so much nicer than the string concat, and less error prone. Wrong nesting of startObject*/end is as easy to achieve here as the old version, but that is nothing in comparison.

File: src/main/java/com/maxmind/geoip2/record/AbstractRecord.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package com.maxmind.geoip2.record;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.IOException;
+
+public class AbstractRecord {
```
Comment (martijnvg) [inline src/main/java/com/maxmind/geoip2/record/AbstractRecord.java]: maybe make this class abstract too? I think no instances will be created of this class, but rather its subclasses?

File: src/main/java/com/maxmind/geoip2/record/AbstractRecord.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    /**
+     * @return JSON representation of this object. The structure is the same as
+     * the JSON provided by the GeoIP2 web service.
+     * @throws IOException if there is an error serializing the object to JSON.
+     */
+    public String toJson() throws IOException {
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
```
Comment (martijnvg) [inline src/main/java/com/maxmind/geoip2/record/AbstractRecord.java]: maybe also add?:

```java
mapper.configure(MapperFeature.CAN_OVERRIDE_ACCESS_MODIFIERS, false);
```

File: src/main/java/com/maxmind/geoip2/record/AbstractRecord.java (hunk 2 above)
Comment (oschwald) [inline src/main/java/com/maxmind/geoip2/record/AbstractRecord.java]: Woops, good catch. I meant to do that.

File: src/main/java/com/maxmind/geoip2/record/AbstractRecord.java (hunk 3 above)
Comment (oschwald) [inline src/main/java/com/maxmind/geoip2/record/AbstractRecord.java]: Good idea.

File: src/main/java/com/maxmind/geoip2/record/AbstractNamedRecord.java [hunk 4]
```diff
@@ -10,13 +11,17 @@
...
  */
-public abstract class AbstractNamedRecord {
+public abstract class AbstractNamedRecord extends AbstractRecord {
 
     private final Map<String, String> names;
     private final Integer geoNameId;
     private final List<String> locales;
 
-    AbstractNamedRecord(Map<String, String> names, Integer geoNameId, List<String> locales) {
+    AbstractNamedRecord() {
```
Comment (eilara) [inline src/main/java/com/maxmind/geoip2/record/AbstractNamedRecord.java]: Who requires all these 0 args constructors? Is it what he JSON lib wants?

Comment (oschwald) [inline src/main/java/com/maxmind/geoip2/record/AbstractNamedRecord.java]: I don't know of anyone that requires them, but removing them would definitely be a breaking change under Semantic Versioning.

**Pull request merged**